instrumentation/azure-functions/src/main/java/io/opentelemetry/instrumentation/auto/azure/functions/AzureFunctionsInstrumentation.java

```diff
@@ -86,5 +91,25 @@ public static Scope methodEnter(@Advice.Argument(0) final Object request)
     public static void methodExit(@Advice.Enter final Scope scope) {
       scope.close();
     }
+
+    /**
+     * Azure Function Linux with consumption plan doesn't come with 'APPLICATIONINSIGHTS_CONNECTION_STRING' env variable during the initialization.
+     * Need to set this connection string lazily when it becomes available.
+     */
+    private static void lazilySetConnectionString() {
+      if (!AiConnectionString.hasConnectionString()) {
```

Comment:
just a comment here
```suggestion
      // race condition (two initial requests happening at the same time) is not a worry here
      // because at worst they both enter the condition below and update the connection string
      if (!AiConnectionString.hasConnectionString()) {
```